Add macOS ARM build platform

mdpiper opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

mdpiper committed:
Use CMAKE_ARGS variable

mdpiper committed:
Bump build number

mdpiper committed:
Build on Windows

mdpiper committed:
Build on osx_arm64

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

mdpiper commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.41.1, and conda-forge-pinning 2024.10.06.14.35.38

mdpiper committed:
Tweak to build on Windows

mdpiper committed:
Revert "Tweak to build on Windows"

This reverts commit 082c01fae142b7c616051f6db1de28bde5591de7.
Windows is not my friend.

mdpiper committed:
Don't bother building on Windows

There's a compile-time error raised by flang, but not by gfortran,
in the GIPL source code. I don't have the bandwidth to go figure
this out.

mdpiper commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.41.1, and conda-forge-pinning 2024.10.06.14.35.38
